x-pack/plugins/observability/server/lib/rules/slo_burn_rate/executor.ts

```diff
@@ -0,0 +1,122 @@
+  ALERT_REASON,
+} from '@kbn/rule-data-utils';
+
+import { Duration, toDurationUnit } from '../../../domain/models';
+import { RulesServiceSetup } from '../rules_service';
+import { DefaultSLIClient, KibanaSavedObjectsSLORepository } from '../../../services/slo';
+import { computeBurnRate } from '../../../domain/services';
+
+export enum AlertStates {
+  OK,
+  ALERT,
+  NO_DATA,
+  ERROR,
+}
+
+export type BurnRateRuleParams = {
+  sloId: string;
+  threshold: number;
+  longWindow: { duration: number; unit: string };
+  shortWindow: { duration: number; unit: string };
+} & Record<string, any>;
+export type BurnRateRuleTypeState = RuleTypeState & {};
+export type BurnRateAlertState = AlertState;
+export type BurnRateAlertContext = AlertContext;
+export type BurnRateAllowedActionGroups = ActionGroupIdsOf<typeof FIRED_ACTION>;
+
+const SHORT_WINDOW = 'SHORT_WINDOW';
+const LONG_WINDOW = 'LONG_WINDOW';
+
+export const createSloBurnRateRuleExecutor = (rulesService: RulesServiceSetup) =>
+  rulesService.createLifecycleRuleExecutor<
+    BurnRateRuleParams,
+    BurnRateRuleTypeState,
+    BurnRateAlertState,
+    BurnRateAlertContext,
+    BurnRateAllowedActionGroups
+  >(async function executor({ services, params, state, startedAt, logger }): Promise<void> {
+    const {
+      alertWithLifecycle,
+      savedObjectsClient: soClient,
+      scopedClusterClient: esClient,
+      alertFactory,
+    } = services;
+
+    const sloRepository = new KibanaSavedObjectsSLORepository(soClient);
+    const sliClient = new DefaultSLIClient(esClient.asCurrentUser);
+    const slo = await sloRepository.findById(params.sloId);
+
+    const longWindowDuration = new Duration(
+      params.longWindow.duration,
+      toDurationUnit(params.longWindow.unit)
+    );
+    const shortWindowDuration = new Duration(
+      params.shortWindow.duration,
+      toDurationUnit(params.shortWindow.unit)
+    );
+
+    const sliData = await sliClient.fetchSLIDataFrom(slo, [
+      { name: LONG_WINDOW, duration: longWindowDuration },
+      { name: SHORT_WINDOW, duration: shortWindowDuration },
+    ]);
+
+    const longWindowBurnRate = computeBurnRate(slo, sliData[LONG_WINDOW]);
+    const shortWindowBurnRate = computeBurnRate(slo, sliData[SHORT_WINDOW]);
+
+    const shouldAlert =
+      longWindowBurnRate >= params.threshold && shortWindowBurnRate >= params.threshold;
+    if (shouldAlert) {
+      const reason = `The burn rate for the past ${longWindowDuration.format()} is ${longWindowBurnRate} and for the past ${shortWindowDuration.format()} is ${shortWindowBurnRate}. Alert when above ${
+        params.threshold
+      } for both window`;
+
+      const context = {
+        longWindow: { burnRate: longWindowBurnRate, duration: longWindowDuration.format() },
+        reason,
+        shortWindow: { burnRate: shortWindowBurnRate, duration: shortWindowDuration.format() },
+        threshold: params.threshold,
+        timestamp: startedAt.toISOString(),
```

Comment:
I'm wondering if we should already add things like `viewInAppUrl` and `alertDetailsUrl` here now as in the other rule types? wdyt?